File: experiments/debug_ver8a.js
// Проверяем основные проблемы с ver8a

console.log('=== VER8A DEBUG TEST ===');

// 1. Проверяем наличие основных элементов
function checkElements() {
    const elements = [
        'rdf-input',
        'input-format', 
        'output-format',
        'layout-engine',
        'visualization-mode',
        'sparql-mode',
        'visualize-btn',
        'max-label-length',
        'max-vad-row-length',
        'smart-design-container',
        'result-sparql-panel',
        'sparql-panel'
    ];
    
    console.log('=== ELEMENT CHECK ===');
    elements.forEach(id => {
        const element = document.getElementById(id);
        console.log(`${id}: ${element ? 'EXISTS' : 'MISSING'}`);
        if (element) {
            console.log(`  - display: ${window.getComputedStyle(element).display}`);
            console.log(`  - visible: ${element.offsetWidth > 0 && element.offsetHeight > 0}`);
        }
    });
}

// 2. Проверяем наличие функций
function checkFunctions() {
    const functions = [
        'visualize',
        'clearRdfInput',
        'saveAsFile', 
        'testRdfValidation',
        'loadExampleTrigVADv4',
        'updateModeDescription',
        'toggleSparqlPanel'
    ];
    
    console.log('\n=== FUNCTION CHECK ===');
    functions.forEach(funcName => {
        const exists = typeof window[funcName] === 'function';
        console.log(`${funcName}: ${exists ? 'EXISTS' : 'MISSING'}`);
    });
}

// 3. Проверяем текущие значения
function checkCurrentValues() {
    console.log('\n=== CURRENT VALUES ===');
    
    const vizMode = document.getElementById('visualization-mode');
    const sparqlMode = document.getElementById('sparql-mode');
    
    if (vizMode) console.log(`visualization-mode: ${vizMode.value}`);
    if (sparqlMode) console.log(`sparql-mode: ${sparqlMode.value}`);
    
    // Проверяем видимость полей
    const maxVadGroup = document.getElementById('max-vad-row-length-group');
    if (maxVadGroup) {
        console.log(`max-vad-row-length-group display: ${maxVadGroup.style.display}`);
    }
    
    const smartDesignContainer = document.getElementById('smart-design-container');
    if (smartDesignContainer) {
        console.log(`smart-design-container classes: ${smartDesignContainer.className}`);
    }
}

// Запускаем проверки
document.addEventListener('DOMContentLoaded', function() {
    setTimeout(() => {
        checkElements();
        checkFunctions();
        checkCurrentValues();
        
        // Пробуем вызвать updateModeDescription
        if (typeof window.updateModeDescription === 'function') {
            console.log('\n=== CALLING updateModeDescription ===');
            window.updateModeDescription();
            
            // Проверяем результат
            setTimeout(() => {
                checkCurrentValues();
            }, 100);
        }
    }, 1000);
});

File: debug_ver8a.py
#!/usr/bin/env python3
"""
Debug script to test ver8a functionality programmatically
"""

import requests
import time
from urllib.parse import urljoin

def test_page_load():
    """Test if page loads correctly"""
    try:
        response = requests.get('http://localhost:8000/index.html')
        if response.status_code == 200:
            print("✓ Page loads successfully")
            return True
        else:
            print(f"✗ Page load failed: {response.status_code}")
            return False
    except Exception as e:
        print(f"✗ Error loading page: {e}")
        return False

def test_js_functions():
    """Test if JavaScript functions exist"""
    try:
        response = requests.get('http://localhost:8000/index.html')
        content = response.text
        
        functions_to_check = [
            'function visualize()',
            'function clearRdfInput()',
            'function saveAsFile()',
            'function testRdfValidation()',
            'function loadExampleTrigVADv4()',
            'function updateModeDescription()',
            'function toggleSparqlPanel()'
        ]
        
        for func in functions_to_check:
            if func in content:
                print(f"✓ {func} found")
            else:
                print(f"✗ {func} missing")
                
        return True
    except Exception as e:
        print(f"✗ Error checking functions: {e}")
        return False

def test_elements():
    """Test if HTML elements exist"""
    try:
        response = requests.get('http://localhost:8000/index.html')
        content = response.text
        
        elements_to_check = [
            'id="rdf-input"',
            'id="visualize-btn"',
            'id="visualization-mode"',
            'id="sparql-mode"',
            'id="max-vad-row-length-group"',
            'id="smart-design-container"',
            'id="result-sparql-query"'
        ]
        
        for element in elements_to_check:
            if element in content:
                print(f"✓ {element} found")
            else:
                print(f"✗ {element} missing")
                
        return True
    except Exception as e:
        print(f"✗ Error checking elements: {e}")
        return False

def main():
    print("=== Debug ver8a Functionality ===")
    print("Testing page at http://localhost:8000/index.html")
    print()
    
    # Wait a moment for server to be ready
    time.sleep(1)
    
    tests = [
        ("Page Load", test_page_load),
        ("JavaScript Functions", test_js_functions),
        ("HTML Elements", test_elements)
    ]
    
    results = []
    for test_name, test_func in tests:
        print(f"\n--- {test_name} ---")
        result = test_func()
        results.append((test_name, result))
    
    print("\n=== Summary ===")
    for test_name, result in results:
        status = "PASS" if result else "FAIL"
        print(f"{test_name}: {status}")

if __name__ == "__main__":
    main()